ExtraRoles/PerformKillPatch.cs

```diff
@@ -24,49 +49,88 @@ public static bool Prefix()
             if (PlayerControl.LocalPlayer.Data.IsDead)
                 return false;
 
-            MessageWriter writer;
             if (CurrentTarget != null)
             {
                 var target = CurrentTarget;
                 //code that handles the ability button presses
                 if (PlayerControl.LocalPlayer.isPlayerRole("Officer"))
                 {
-                    if (PlayerTools.getOfficerCD() != 0)
+                    if (PlayerTools.getOfficerCD() > 0)
+                    {
                         return false;
-                    
-                    var isPlayerJokerAndCanDieToOfficer =
-                        Main.Config.jokerCanDieToOfficer && target.isPlayerRole("Joker");
-                    
-                    writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId,
-                        (byte) CustomRPC.OfficerKill, Hazel.SendOption.None, -1);
-                    writer.Write(PlayerControl.LocalPlayer.PlayerId);
-                    writer.Write(!target.isPlayerImmortal() &&
-                        (isPlayerJokerAndCanDieToOfficer || target.Data.IsImpostor) ?
-                        target.PlayerId : PlayerControl.LocalPlayer.PlayerId); // kill target or suicide
-                    AmongUsClient.Instance.FinishRpcImmediately(writer);
-                    PlayerControl.LocalPlayer.MurderPlayer(target);
-                    PlayerControl.LocalPlayer.getModdedControl().LastAbilityTime = DateTime.UtcNow;
+                    }
 
                     // attempted to kill shielded player
                     if (target.isPlayerImmortal())
                     {
-                        writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId,
-                            (byte) CustomRPC.AttemptSound, Hazel.SendOption.None, -1);
-                        AmongUsClient.Instance.FinishRpcImmediately(writer);
+                        // suicide packet
+                        WriteKillRpc(PlayerControl.LocalPlayer, PlayerControl.LocalPlayer);
                         BreakShield(false);
+                        return false;
                     }
-                    
+
+                    var officerKillSetting = (OfficerKillBehaviour) Main.Config.officerKillBehaviour;
+                    if (officerKillSetting == OfficerKillBehaviour.OfficerSurvives)
```

Comment:
Don't need to return here, nor have multiple statements checking for all the conditions in which the target will die.
```csharp
if (!target.isPlayerImmortal() && // not shielded
    (officerKillSetting == OfficerKillBehaviour.OfficerSurvives // don't care who it is, kill them
    || isTargetImpostor // impostors always die
    || officerKillSetting == OfficerKillBehaviour.Joker && target.isPlayerRole("Joker"))) // joker can die and target is joker
{
    // kill target
}
else // officer dies
{
    if (officerKillSetting == OfficerKillBehaviour.CrewDie)
    {
        // kill target too
    }
    // kill officer
}

// attempted to kill shielded player
if (target.isPlayerImmortal())
    BreakShield(false);

return false;
```

Reply:
I am really not a fan of complex conditions. I prefer the readability over shorter code. Also your suggestion includes the wrong behaviour of an officer attacking a shielded player.

Reply:
It was only wrong in the case of killing crewmates, which was my mistake, could've been fixed by adding `!target.isPlayerImmortal() &&` infront of the setting comparison.
Also the condition is not complex, much less repetitive and in my opinion much more readable too, which is cleaner and can sometimes lead to better performance too depending on application. I even added comments, which I usually don't even in my code because I find these conditions very easy to read in most cases.